src/graphnet/models/components/layers.py

```diff
@@ -59,3 +65,140 @@ def forward(
+
+    'IceCube - Neutrinos in Deep' kaggle competition.
+    """
+
+    def __init__(
+        self,
+        layer_sizes: Optional[List[int]] = None,
+        aggr: str = "max",
+        nb_neighbors: int = 8,
+        features_subset: Optional[Union[Sequence[int], slice]] = None,
+        **kwargs: Any,
+    ):
+        """Construct `DynTrans`.
+
+        Args:
+            nn: The MLP/torch.Module to be used within the `DynTrans`.
+            layer_sizes: List of layer sizes to be used in `DynTrans`.
+            aggr: Aggregation method to be used with `DynTrans`.
+            nb_neighbors: Number of neighbours to be clustered after the
+                `EdgeConv` operation.
+            features_subset: Subset of features in `Data.x` that should be used
+                when dynamically performing the new graph clustering after the
+                `EdgeConv` operation. Defaults to all features.
+            **kwargs: Additional features to be passed to `DynTrans`.
+        """
+        # Check(s)
+        if features_subset is None:
+            features_subset = slice(None)  # Use all features
+        assert isinstance(features_subset, (list, slice))
+
+        if layer_sizes is None:
+            layer_sizes = [256, 256, 256]
+        layers = []
+        for ix, (nb_in, nb_out) in enumerate(
+            zip(layer_sizes[:-1], layer_sizes[1:])
+        ):
+            if ix == 0:
+                nb_in *= 3  # edgeConv1
+            layers.append(torch.nn.Linear(nb_in, nb_out))
+            layers.append(torch.nn.LeakyReLU())
+        d_model = nb_out
+
+        # Base class constructor
+        super().__init__(nn=torch.nn.Sequential(*layers), aggr=aggr, **kwargs)
+
+        # Additional member variables
+        self.nb_neighbors = nb_neighbors
+        self.features_subset = features_subset
+
+        self.norm_first = False
+
+        self.norm1 = LayerNorm(d_model, eps=1e-5)  # lNorm
+
+        # Transformer layer(s)
+        encoder_layer = TransformerEncoderLayer(
+            d_model=d_model,
+            nhead=8,
+            batch_first=True,
+            norm_first=self.norm_first,
+        )
+        self._transformer_encoder = TransformerEncoder(
+            encoder_layer, num_layers=1
+        )
+
+    def forward(
+        self, x: Tensor, edge_index: Adj, batch: Optional[Tensor] = None
+    ) -> Tensor:
+        """Forward pass."""
+        if self.norm_first:
+            x = self.norm1(x)  # lNorm
+
+        x_out = super().forward(x, edge_index)
+
+        if x_out.shape[-1] == x.shape[-1]:
+            x = x + x_out
+        else:
+            x = x_out
+
+        if not self.norm_first:
+            x = self.norm1(x)  # lNorm
+
+        # Recompute adjacency
+        edge_index = None
+
+        # Transformer layer
+        x, mask = to_dense_batch(x, batch)
+        x = self._transformer_encoder(x, src_key_padding_mask=~mask)
+        x = x[mask]
+
+        return x, edge_index
```

Comment:
The new `edge_index` is set to `None`; do we even need to return it? If not, please consider removing this

Reply:
Removed, also from the GNN constructor.